src/playlist-selectors.js

```diff
@@ -375,5 +375,5 @@ export const lowestBitrateCompatibleVariantSelector = function() {
   const playlistsWithVideo =
     playlists.filter(playlist => parseCodecs(playlist.attributes.CODECS).videoCodec);
 
-  return playlistsWithVideo[0] || playlists[0];
+  return playlistsWithVideo[0] || null;
```

Comment:
If possible I'd like to avoid coupling this function to a separate algorithm, and allow the consumer to decide the alternative.  I think long term adding the complexity here over elsewhere will be a bit confusing, as well as making unit testing and changes harder/more complex.